### DISCUSSION TITLE
Use existing domain and bypass FreeDNS?

### DISCUSSION THREAD

**Discussion by robschwieb** [NightscoutFoundation/xDrip#2569] [Q&A]
Amazing work you guys have done here. Just went through the setup process and I'm really impressed. I've long envisioned something like this as Heroku alternative but don't have the skills to do so.

Anyway, I've been self-hosting my son's NS and would like to transition to this configuration for a few reasons (power outages this summer, and the non-static nature of residential IP addresses).

I already have a google domain that I pay $12 a year for and would like to use that instead of the FreeDNS for ease of transitioning and so we don't have to update his doctors or anyone else who relies on it.

I figured I could just use the VM instance's external IP in place of my home network external IP in the Google Domains config. However, this didn't work? I was able to access it by `www.my-custom-domain.com` or `http://my-custom-domain.com` but not `my-custom-domain.com` or `https://my-custom-domain.com`.

I'm by no means a network guy and know just enough to be dangerous. I'm sure I'm missing something obvious and any help in pointing it out would be greatly appreciated!

**Comment by Navid200** [comment]
I would like to add the option of using a private domain as a part of the installation routine.
I may ask you to test it for us.

**Reply by robschwieb** [reply]
That would be great and I'd be more than happy to help test. FWIW, my current install that I self host is made up of dockers. I'm currently using Nginx Proxy Manager to generate SSL Certs via Let's Encrypt and it's been flawless.

**Comment by tzachi-dar** [comment]
If you can use http://my-custom-domain.com but not https://my-custom-domain.com
please run:
 sudo certbot --nginx -d my-custom-domain.com --redirect --agree-tos --no-eff-email

**Reply by Navid200** [reply]
Is this something that can be executed right after completing installation phase 1?

Or does the user first have to create a FreeDNS account and add a subdomain and complete installation phase 2 before running this?

**Comment by tzachi-dar** [comment]
I think he will have to run installation phase 2.
If he is lucky, it might work after he gets to the freedns dialog. He can try pressing control c then, but not before that.

@robschwieb Do you have a solution to the issue of the machine addresses changing after a reboot?

**Reply by robschwieb** [reply]
Machine address changing after reboot? Are you referring to my original gripe about residential internet service? If so, I just remote into one of my LAN machines using Google Remote access and see what my external IP changed to. Then I update the Google Domain to point to that new address.

**Comment by robschwieb** [comment]
Any other suggestions or thoughts on this? It's the only piece I'm missing before making the transition.

**Reply by Navid200** [reply]
We have limited manpower, who are all volunteers.
We will not survive if we don't focus.  Our objective is to provide an option for the masses, a free option mind you.
Extreme customization is not something we can dedicate time to.

Even if I was 100% convinced that using a private domain was advantageous over using a FreeDNS domain that not many others are using, it would still not fit in our objective of having a free solution as a private domain is not free.

We already have open PRs that have not been merged.  Those are holding up additional work that needs to fall into place.
Now, we are looking at sorting out the option to implement the next release of Nightscout.
There is a lot of work we have to take care of.

There are projects I need to work on that are related to xDrip.  I have not touched xDrip for 4 months now because of the time this has taken.

I am afraid what you are asking for may not get the attention it needs in order to be implemented.
Considering our limited manpower, we need to prioritize.  And private domain I don't believe should get priority.  At least, not right now.

**Reply by robschwieb** [reply]
Completely understand and agree.

I'll pursue a different route but could you she'd any light on how your
automated NS database transfer works? I have another docker based solution
in a cloud VM but didn't finish it due to being unable to easily migrate my
son's data.

Your tool made it too easy and wish it was available as a standalone
package.
